packages/core/rfcs/20200401-products-filters.md

```diff
@@ -0,0 +1,57 @@
+# Filtering products
+
+Filtering products is an important part of any products listing page. To ensure the best developer experience and consistency between integrations (because filters have to be delivered to the theme in a unified form) we need to provide a common way to retrieve available filters and unified interface for them.
+
+## New APIs
+
+### Modify `useProduct` composable
+
+Since there are several factors that the filters depend on (products type, category, available products, their attributes) and the set of filters might be dynamic, this feature should introduce new field in the `useProduct` composable, `availableFilters`, like in the example below.
+
+**New interface**:
+
+```ts
+export interface UseProduct<PRODUCT, FILTER_OPTION> {
+  products: ComputedProperty<PRODUCT[]>;
+  totalProducts: ComputedProperty<number>;
+  availableFilters: ComputedProperty<Filter<FILTER_OPTION>[]>;
```

Comment:
IMHO this field should be optional for platforms that don't send available filters.